.github/workflows/check-and-gen.yml

```diff
@@ -70,3 +68,37 @@ jobs:
       - uses: ./.github/actions/install-go
       - run: go mod tidy -v
       - run: git diff --exit-code
+  generate-docs:
+    name: Generate Documentation
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release/'))
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          fetch-depth: 0
+
+      - uses: ./.github/actions/install-system-dependencies
+      - uses: ./.github/actions/install-go
+      - uses: ./.github/actions/make-deps
+
+      - name: Generate API documentation using docsgen-cli
+        run: |
+          make docsgen-cli || {
+            echo "Error: Documentation generation failed"
+            exit 1
+          }
+
+      - name: Commit and push if documentation changed
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            git config user.name "github-actions[bot]"
+            git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
I have no expertise here, but maybe add a comment on where this comes from?  

Is https://github.com/orgs/community/discussions/26560 the canonical link?

Reply:
Yeah that's correct, I'll add a comment pointing to the GitHub discussion for future references